import "ModID.proto";
package  Login;

enum LOGIN_MOD
{
	ReqLogin_req = 1;				//请求登录
	RetLoginSucess_rsp = 2;		//登录成功
	LoginError_rsp = 3;				//登录错误
}

//请求登陆
message ReqLogin
{
	optional string  account = 1;		//帐号
	optional string  password = 2;		//密码
	
	enum ProtoModID
	{
		ModID = 1;
	}
	
	enum ProtoFunID
	{
		FunID = 1;	
	}
}

//返回登陆结果
message RetLoginSucess
{
	optional string token =1;				//登陆token
	optional string ip = 2;
	optional string device = 3;
	optional uint32 opID = 4;

	enum ProtoModID
	{
		ModID = 1;
	}
	
	enum ProtoFunID
	{
		FunID = 2;	
	}

}

//返回登录错误

message LoginError
{
	enum ErrorType
	{
		PASSWORD_WRONG = 1;	//密码错误
		HAS_LOGIN_IN  = 2;	//已经登录了
		OTHER = 3;			//其它原因
		LOGIN_IN_OTHER_PLACE = 4;//在其它地方登录
		LOGIN_VERSION_TOO_LOW = 5;	//客户端版本太低
	}
	optional uint32  errorType = 1;
	
	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 3;	
	}
}

//请求登录网关
message ReqLoginGate
{
	optional string token = 1;			//从登录服务器获得的token
	optional string deviceID = 2;		//设备ID
	optional string ip = 3;				//客户端ip
	optional uint32 opID = 4;			//运营商ID
	optional uint32 clientVersion = 5;	//客户端版本
	optional uint32 serverID = 6;		//serverID
	optional string extraData  = 7;		//sdk验证 json 格式
	optional uint32 sdkId = 8;          //sdk id

	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 4;	
	}
}

//请求服务器时间 c --->s 
message stReqServerTime
{
	optional uint32 flag = 1;		//随便填一个大于0的数
	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 5;	
	}
}

//返回服务器时间 s ---> c
message stRetServerTime
{
	optional uint32 svrTime = 1;		//服务器时间

	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 6;	
	}
}

//请求重新连接 c -----> s
message stAutoLoginGate
{
	optional string token = 1;		//从登录服务器获得的token
	optional string deviceID = 2;	//设备ID
	optional string ip = 3;			//客户端ip
	optional uint32 opID = 4;		//运营商ID
	optional string extraData  = 5;	//sdk验证 json 格式
	optional uint32 sdkId = 6;      //sdk id

	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 7;	
	}
}

//要重新登录 s ----> c
message stNotifyRelogin
{
	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 8;	
	}
}

//客户端的ping值查询 c <----------------> s
message stQueryPingValue
{
	optional uint32 curTime = 1;		//当前的客户端值,客户端发送时候的时间，毫秒为单位

	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 9;	
	}
}

//服务器请求客户端的时间  s -------> c
message stReqClientTime
{
	optional uint32 curTime = 1;

	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 10;	
	}
}

//发送自己的时间给服务器 c -------> s
message stSendClient2Svr
{
	optional uint32 curTime = 1;

	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 11;	
	}
}

//是否是新建账号 s -------> c
message stNotifyIsNewAccount
{
	optional bool isNewAcc = 1;     //是否是新用户
    optional string account = 2;    //登录名

	enum ProtoID
	{
		ModID = 1;
	}

	enum ProtoFunID
	{
		FunID = 12;	
	}
}
